src/test/ui/emit-directives.rs

```diff
@@ -0,0 +1,6 @@
+// A very basic test for the emission of build directives in JSON output.
+
+// compile-flags:--emit=metadata --error-format=json -Z emit-directives
```

Comment:
Is `--error-format=json` also required here? Reading the code it didn't appear to be so but I'm not entirely sure

Reply:
It is. The default impl for `maybe_emit_json_directive` is empty. `JsonEmitter` is the only type that implements a non-empty version, and it's only used when `--error-format=json` is specified.

This seems desirable -- we don't want to allow output that's a mixture of JSON and non-JSON.